server/src/services/metadata.service.ts

```diff
@@ -428,7 +428,7 @@ export class MetadataService {
 
     if (isMotionPhoto && directory) {
       for (const entry of directory) {
-        if (entry.Item.Semantic == 'MotionPhoto') {
+        if (entry?.Item?.Semantic == 'MotionPhoto') {
```

Comment:
`entry` should never be null or undefined, should it?

Reply:
We know that `directory` is an array from an above `Array.isArray` check, but we don't know what it is an array of, I think technically it could be an array of null values, although I don't know if that is practically possible or not. Basically, this is technically valid, so I just added it anyways.
```ts
> const directory = [null]
undefined
> Array.isArray(directory)
true
> directory[0].Item
Uncaught TypeError: Cannot read properties of null (reading 'Item')
> 
```